src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -173,22 +173,29 @@ public IDataReader ExecuteQuery(string query, CancellationToken cancellationToke
             clientRequestProperties.SetOption(ClientRequestProperties.OptionNoTruncation, true);
             cancellationToken.Register(() => CancelQuery(clientRequestProperties.ClientRequestId));
 
-            var kustoCodeService = new KustoCodeService(query);
-            var minimalQuery = kustoCodeService.GetMinimalText(MinimalTextKind.RemoveLeadingWhitespaceAndComments);
+            var script = CodeScript.From(query, GlobalState.Default);
+            IDataReader[] origReaders = new IDataReader[script.Blocks.Count];
 
-            try
+            Parallel.ForEach(script.Blocks, (codeBlock, state, index) =>
```

Comment:
Should we pass the cancellation token into this Parallel.Foreach incase the user cancels?

Reply:
Looks like in previous implementation, its not passed. So, it should not regress anything. We can discuss and target that in a separate PR too.